distributed/client.py

```diff
@@ -3040,7 +3040,7 @@ def _expand_key(cls, k):
             k = (k,)
         for kk in k:
             if dask.is_dask_collection(kk):
-                for kkk in kk.__dask_keys__():
+                for kkk in list(flatten([kk])):
```

Comment:
You probably want to continue using `kk.__dask_keys__()` like you were before.  The `flatten` function is there just to collapsed the nested list of lists to a single list.

Reply:
And sorry for the delay on review.  I appreciate your contribution here and your patience.

Reply:
No problem at all. I'm amazed that you can keep track of everything going on on this project!
My new test is still failing because it is looking for a key for the dask.array.from_delayed() object. However, I can't figure out what that key looks like in worker.data. Any tip on how to check that?

Reply:
Keys get transformed by `distributed.utils.tokey` which mostly just converts them into strings.